operator/internal/handlers/internal/tlsprofile/tlsprofile.go

```diff
@@ -0,0 +1,71 @@
+package tlsprofile
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	projectconfig_types "github.com/grafana/loki/operator/apis/config/v1"
+	"github.com/grafana/loki/operator/internal/external/k8s"
+	openshiftv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/crypto"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetSecurityProfileInfo gets the tls profile info to apply.
+func GetSecurityProfileInfo(ctx context.Context, k k8s.Client, tlsProfileType projectconfig_types.TLSProfileType) (projectconfig_types.TLSProfileSpec, error) {
```

Comment:
Please add some tests for this functions to safe guard that we extract the spec correctly for the supported cases.

Reply:
was just working on that